app/controllers/users_controller.rb

```diff
@@ -19,4 +21,29 @@ def other_instances
     tenant = current_tenant
     ActsAsTenant.without_tenant { Instance.containing_user(@user) - [tenant] }
   end
+
+  def destroy
+    if @course_user.student?
+      @course_user.destroy
+      respond_to do |format|
+        format.html { redirect_to course_users_path(@course), notice: 'Student was successfully removed from the course.' }
+        format.json { head :no_content }
+      end
+    else
+      respond_to do |format|
+        format.html { redirect_to course_users_path(@course), alert: 'Only students can be removed.' }
+        format.json { render json: { error: 'Only students can be removed' }, status: :unprocessable_entity }
+      end
+    end
+  end
+
+  private
+
+  def set_course
+    @course = Course.find(params[:course_id])
+  end
+
+  def set_course_user
+    @course_user = @course.course_users.find(params[:id])
+  end
```

Comment:
Your logic should be modify `Course::UsersController`, not top-level `UsersController`.  
You should not be modifying User.

I just saw that you removed this change in the next commit. You should squash this changes as I review commit-by-commit.